src/options.c

```diff
@@ -369,30 +399,86 @@ JOHN_USAGE_REGEX \
 "                           tunable cost parameters, see doc/OPTIONS\n" \
 "--save-memory=LEVEL        enable memory saving, at LEVEL 1..3\n" \
 "--node=MIN[-MAX]/TOTAL     this node's number range out of TOTAL count\n" \
+"--format=CLASS             valid classes: dynamic, cpu\n" \
```

Comment:
We already listed `--format=NAME`. Now when we drop hidden options that one should be `--list=[NAME|CLASS][,...]`

Reply:
BTW I see we forgot to update this. Current valid classes are `dynamic, cpu, opencl, ztex, mask, omp, all, enabled, disabled`. So something more like
```
"--format=[NAME|CLASS]      force hash type. The supported formats can be seen\n" \
"                           with --list=formats and --list=subformats. Valid\n" \
"                           classes: dynamic, cpu, opencl, ztex, mask, omp, all,\n" \n
"                           enabled, disabled.\n"
```

Reply:
I'll refrain from changing this in #4395 to avoid merge problems. Please make my suggested change above!

Reply:
Perfect, I applied the change to the `--format=` macro. However, it was unclear to me whether I should make any changes to `--list`?

Reply:
No that was a typo of mine. I meant `--format=[NAME|CLASS][,...]`

Reply:
Hmm so maybe I also meant to include `[,...]` below but we're one character short so it'll just be `[,..]`
```
"--format=[NAME|CLASS][,..] force hash type. The supported formats can be seen\n" \
"                           with --list=formats and --list=subformats. Valid\n" \
"                           classes: dynamic, cpu, opencl, ztex, mask, omp, all,\n" \
"                           enabled, disabled.\n"
```

Reply:
Applied.